apps/host/src/components/modules/challenges/challenge-grid/challenge-grid.tsx

```diff
@@ -17,19 +17,24 @@ interface Props {
 function ChallengeGrid({ challenges, linkPrefix, links }: Props) {
   const [parent] = useAutoAnimate();
   const [searchInput, setSearchInput] = useState<string>('');
+  const [newChallenges, setNewChallenges] = useState<boolean>(false);
   const [sortedChallenges, setSortedChallenges] = useState(() =>
     getSortedChallengesByDifficulty(challenges)
   );
 
   useEffect(() => {
+    let filteredChallenges: IChallenge[] = [];
     if (searchInput) {
-      setSortedChallenges(() =>
-        filterChallengeByKey([...challenges.values()], 'title', searchInput)
-      );
+      const filteredByTitle = filterChallengeByKey([...challenges.values()], 'title', searchInput);
+      filteredChallenges = newChallenges ? filterChallengeByKey([...filteredByTitle.values()], "isNew", true) : filteredByTitle;
+    } else if (newChallenges) {
+      filteredChallenges = filterChallengeByKey([...challenges.values()], "isNew", true);
     } else {
-      setSortedChallenges(() => getSortedChallengesByDifficulty(challenges));
+      filteredChallenges = getSortedChallengesByDifficulty(challenges);
     }
-  }, [challenges, searchInput]);
+
+    setSortedChallenges(filteredChallenges);
+  }, [challenges, searchInput, newChallenges]);
```

Comment:
doing it in multi-level, will that be efficient & cleaner?

filteredChallenges = filterByTitle(challenges, text)
newChallenges = filterByNew(filteredChallenges, isNew)

Reply:
Hey @sadanandpai , not sure I follow you. Could you please elaborate the point here again?

Reply:
in steps I meant
`filteredChallenges`
`newChallenges`

if that helps to write cleaner code